src/charm.py

```diff
@@ -275,6 +313,18 @@ def _restart(self, event: EventBase) -> None:
 
         self.container.restart(CONTAINER)
 
+        if broker_active(
+            unit=self.unit,
+            zookeeper_config=self.kafka_config.zookeeper_config,
+        ):
+            logger.info(f'Broker {self.unit.name.split("/")[1]} restarted')
+            self.unit.status = ActiveStatus()
+        else:
+            self.unit.status = BlockedStatus(
+                f"Broker {self.unit.name.split('/')[1]} failed to restart"
+            )
+            return
+
     @property
     def ready_to_start(self) -> bool:
```

Comment:
Same here. On the [VM code](https://github.com/canonical/kafka-operator/pull/41/files#diff-b9ed39bbc9c0387bd3e07da31d13373745534a1cd723d3e292c73496b12e307cR323-R327) you are checking for ranges to be met on storage options

Reply:
For K8s, there isn't the ability to `add` a storage, we're limited to one, so I don't think we'd need this functionality.